compose/utils.py

```diff
@@ -91,3 +93,15 @@ def json_hash(obj):
     h = hashlib.sha256()
     h.update(dump)
     return h.hexdigest()
+
+
+def interpolate(string, mapping):
+    try:
+        return Template(string).substitute(defaultdict(lambda: "", mapping))
+    except ValueError:
```

Comment:
I've updated the test file so that the "unset variable" case gets its own test, just so it's clear we do handle that case.